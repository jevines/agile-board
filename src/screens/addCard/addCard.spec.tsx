import React from "react";
import { render, fireEvent } from "@testing-library/react";
import AddCard from "./addCard";

const mockTitle = "A Mock Title";
const mockDescription = "A Mock description";
const mockComments = ["mock comment 1"];

const expectedTask = {
  id: expect.any(String),
  title: mockTitle,
  description: mockDescription,
  comments: mockComments,
};

describe("The AddCard screen", () => {
  let container: HTMLElement;
  let getByTestId: Function;

  beforeEach(() => {
    ({ container, getByTestId } = renderAddCard());
  });

  it("should match the snapshot", () => {
    expect(container.firstChild).toMatchSnapshot();
  });

  describe("cliking the submit button", () => {
    beforeEach(() => {
      const titleInput = getByTestId("titleInput");
      fireEvent.change(titleInput, { target: { value: mockTitle } });

      const descriptionInput = getByTestId("descriptionInput");
      fireEvent.change(descriptionInput, {
        target: { value: mockDescription },
      });

      const commentsInput = getByTestId("commentsInput");
      fireEvent.change(commentsInput, { target: { value: mockComments[0] } });

      fireEvent.click(getByTestId("submitCardButton"));
    });

    it("should call the onSubmit callback with the enter data", () => {
      expect(mockOnSubmit).toHaveBeenCalledTimes(1);
      expect(mockOnSubmit).toHaveBeenCalledWith(expectedTask);
    });
    it("should call the onClose callback", () => {
      expect(mockOnClose).toHaveBeenCalledTimes(1);
    });
  });
});

const mockOnSubmit = jest.fn();
const mockOnClose = jest.fn();

const renderAddCard = () => {
  return render(<AddCard onSubmit={mockOnSubmit} onClose={mockOnClose} />);
};
