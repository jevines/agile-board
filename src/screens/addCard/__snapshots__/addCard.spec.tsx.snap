// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The AddCard screen should match the snapshot 1`] = `
<section
  class="sc-bdfBwQ cSqurG"
  data-testid="addCardScreen"
>
  <input
    class="sc-gsTCUz dQuXtS"
    data-testid="titleInput"
    value="title"
  />
  <input
    class="sc-gsTCUz dQuXtS"
    data-testid="descriptionInput"
    value="description"
  />
  <input
    class="sc-gsTCUz dQuXtS"
    data-testid="commentsInput"
    value="comments"
  />
  <button
    class="sc-dlfnbm ceszxr"
    data-testid="submitCardButton"
  >
    Submit
  </button>
</section>
`;
